Aztec/Classes/Inserters/Base/Inserter.swift

```diff
@@ -0,0 +1,46 @@
+import UIKit
+
+protocol Inserter {
+    func insert(for element: ElementNode, inheritedAttributes: [AttributedStringKey: Any]) -> NSAttributedString
+
+    /// The special string that represents the element — usually `NSAttachmentCharacter`.
+    ///
+    /// - Parameters:
+    ///
+    ///     - element: Element that should be represented by the replacement String.
+    func specialString(for element: ElementNode) -> String
+
+    /// Returns an attachment representing the HTML element.
+    ///
+    /// - Parameters:
+    ///
+    ///     - representation: HTML element to turn into an attachment.
+    ///     - inheritedAttributes: Attributes inherited from the parent element — to be turned into `extraAttributes` on the `Attachment` itself.
+    ///
+    /// - Returns: Attachment when appropriate, `nil` when there isn't a valid transformation into an Attachment.
+    func attachment(from element: ElementNode, inheriting inheritedAttributes: [AttributedStringKey: Any]) -> NSTextAttachment?
```

Comment:
I wonder if the result type could be a generic... `<T: NSTextAttachment>`, so that the classess that implement this may customize the result type, and we get away without casting all the things.

Reply:
Two observations... 

1. This is starting to look more and more like a `Converter` than an `Inserter`.  It's not important right now, but we may need to revisit the naming (start thinking!).

2. @jleandroperez you may be right.

Reply:
Doesn't any type information get lost as soon as we actually attach it to the `NSAttributedString`? 

Attributes are typed as `[AttributedStringKey: Any]`, so even if the `Inserter`s returned something more specific than `NSTextAttachment`, the only place where this would be useful is after we created an attachment, but haven't attached it to a string yet — does that buy us anything useful?